packages/flutter_test/lib/src/image.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/painting.dart';
+
+import 'test_async_utils.dart';
+
+final Map<int, ui.Image> _cache = <int, ui.Image>{};
+
+/// Creates an arbitrarily sized image for testing.
+///
+/// If the [cache] parameter is set to true, the image will be cached for the
+/// rest of this suite. This should be avoided for images that are used only
```

Comment:
I was slightly confused by this telling me to avoid it and the fact that it defaults to true. Maybe add a sentence describing why you usually want caching?

Reply:
Why do we need the caching at all? Does re-creating the image really cost that much in terms of testing performance?

Reply:
My thought is that we're going from what is a very fast operation (constructing a fake implementation) to what is now a measurably slower and more memory intensive operation, particularly if the image tested is large.

Most of our test cases use a 10x10 image, completely arbitrarily. Some of them use as big as 200x300. If our CI machines are running slowly, caching that 200x300 could save us something on the order of seconds.

Updated the docs here to explain that.